ui/pages/confirmations/components/confirm/footer/shield-footer-coverage-indicator/shield-footer-coverage-indicator.tsx

```diff
@@ -0,0 +1,94 @@
+import { SignatureRequest } from '@metamask/signature-controller';
+import { TransactionMeta } from '@metamask/transaction-controller';
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
+import { useAlertMetrics } from '../../../../../../components/app/alert-system/contexts/alertMetricsContext';
+import InlineAlert from '../../../../../../components/app/alert-system/inline-alert/inline-alert';
+import { MultipleAlertModal } from '../../../../../../components/app/alert-system/multiple-alert-modal';
+import { ConfirmInfoRow } from '../../../../../../components/app/confirm/info/row';
+import { RowAlertKey } from '../../../../../../components/app/confirm/info/row/constants';
+import { Box, Text } from '../../../../../../components/component-library';
+import {
+  AlignItems,
+  Severity,
+  TextColor,
+  TextVariant,
+} from '../../../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../../../hooks/useI18nContext';
+import {
+  getCoverageStatus,
+  ShieldState,
+} from '../../../../../../selectors/shield/coverage';
+import { useConfirmContext } from '../../../../context/confirm';
+
+const ShieldFooterCoverageIndicator = () => {
+  const t = useI18nContext();
+  const { currentConfirmation } = useConfirmContext<
+    TransactionMeta | SignatureRequest
+  >();
+  const { trackInlineAlertClicked } = useAlertMetrics();
+
+  const coverageStatus = useSelector((state) =>
+    getCoverageStatus(state as ShieldState, currentConfirmation.id),
+  );
+
+  const [alertModalVisible, setAlertModalVisible] = useState<boolean>(false);
+
+  const handleModalClose = () => {
+    setAlertModalVisible(false);
+  };
+
+  const handleInlineAlertClick = () => {
+    setAlertModalVisible(true);
+    trackInlineAlertClicked(RowAlertKey.ShieldFooterCoverageIndicator);
+  };
+
+  return (
+    <Box
+      paddingLeft={4}
+      paddingRight={4}
+      // box shadow to match the original var(--shadow-size-md) on the footer,
+      // but only applied to the top of the box, so it doesn't overlap with
+      // the existing
+      style={{ boxShadow: '0 -4px 16px -8px var(--color-shadow-default)' }}
+    >
+      {alertModalVisible && (
+        <MultipleAlertModal
```

Comment:
Why do we need a manual modal, could this not be automatic based on the new alert?

Could we use an alert row to do some of this for us and get the `InlineAlert` for free?